chart/kubeapps/templates/kubeapps-frontend-config.yaml

```diff
@@ -61,10 +61,29 @@ data:
       location ~* /api/clusters/{{ .name }} {
         rewrite /api/clusters/{{ .name }}/(.*) /$1 break;
         rewrite /api/clusters/{{ .name }} / break;
+
+      # Helm returns a nil pointer error when accessing foo.bar if foo doesn't
+      # exist, even with the `default` function.
+      # See https://github.com/helm/helm/issues/8026#issuecomment-756538254
+      {{- $pinnipedConfig := .pinnipedConfig | default dict }}
+      {{- if and $.Values.pinnipedProxy.enabled $pinnipedConfig.exchangeCredentials }}
```

Comment:
Would it make sense to simplify this and only allow all-or-none configuration regarding pinniped? (just configuring all the clusters with the same pinniped instance if given)

I am okay leaving this as it is since it's already working but the additional flexibility may be adding unnecessary complexity to the chart and config.

Reply:
> Would it make sense to simplify this and only allow all-or-none configuration regarding pinniped? (just configuring all the clusters with the same pinniped instance if given)

You (as the person installing Kubeapps) may not have that option if you are given access to setup Kubeapps with existing clusters, some of which use pinniped, some straight oidc, some different pinniped authenticators etc. So yes, it would be simpler if we had the guarantee that "pinniped-concierge will always be setup with the same namespace and authenticator details for all clusters", but given that we don't, I didn't want to enforce that as a restriction.

> 
> I am okay leaving this as it is since it's already working but the additional flexibility may be adding unnecessary complexity to the chart and config.

I've tried to do this so it doesn't have unnecessary complexity for the user: so the defaults assume that pinniped is install in the same namespace and the same authenticator in all clusters, but can be changed.